### ISSUE TITLE
Bug: LZ is making variations with illegal moves

### ISSUE THREAD

**Issue by pondturtle** [pnprog/goreviewpartner#39]
![screen](https://user-images.githubusercontent.com/28937761/38456623-d319c566-3a86-11e8-9c42-d5e14df7be8f.jpg)

Happens ever since I got latest weights file.

**Comment by pnprog** [comment]
Hi!

Could you try with the latest version of GRP (currently v0.11) and latest version of Leela Zero (v0.13)? Oldest and newest versions of Leela may have different behaviour regarding undoing moves, this could explain you issue.

**Comment by pnprog** [comment]
Do you mean you can easily reproduce that bug? Because, I cannot reproduce it myself, even with the latest weights file, on the same game position as your, with or without move span.

Another user had a somewhat similar issue, but I was not able to reproduce the issue :(

If reproducing this issue is doable on your side, I would like maybe to have you try with a dedicated version of GRP that would produce a lot of log (list of all GTP commands sent by GRP, complete Leela Zero standard error log, plus more detailed GRP log) for me to analyze in details and understand what is happening.

That would help me a lot!

**Comment by pondturtle** [comment]
I would love to help. Please give me comprehensive list of what you need.

**Comment by pnprog** [comment]
I will prepare something.

But first, I need to know: Do you use Windows? or MacOs or Linux? If Windows, do you use the package version? or run from python source?

**Comment by pondturtle** [comment]
Ok I have to apologize profusely here. But I got the folders mixed up and this was actually happening with version 0.10. I am very sorry for any time wasted on your part. I feel very bad about this.

I downloaded latest "goreviewpartner-v0.11-Leela-Zero" package and the problem is gone. Are you still interested or does that render the whole isssue moot? I kept the version which I can use to reproduce the bug in archive if you are interested.
